src/hooks/usePagenation.ts

```diff
@@ -0,0 +1,56 @@
+import { useRef, useState, useEffect } from 'react'
+
+interface usePagenationProps {
+  [key: string]: string
+}
+
+type usePageNation = {
+  currPage: number
+  lastPage: number
+  currPageData: usePagenationProps[]
+  onClickNextPage: () => void
+  onClickPrevPage: () => void
+}
```

Comment:
이 부분은 저의 개인적인 견해긴 한데,
그동안 원석님이 작성하신 코드들 봤을 때 객체 타입은 interface로 작성하시고 그 안에서 한 번 더 짧게 (예를 들면 1줄 정도의) 타입 지정이 필요할 때 type으로 작성하셨던 것 같아서
일관성을 위해서 이 부분도 그렇게 작성해주시면 좋을 것 같아요!

```suggestion
type usePagenation = {
  [key: string]: string
}

interface usePageNationProps  {
  currPage: number
  lastPage: number
  currPageData: usePagenation[]
  onClickNextPage: () => void
  onClickPrevPage: () => void
}

Reply:
아 정확하세요! 앞으로는 한번 더 확인해야겠네요! 수정했습니다.!